apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java

```diff
@@ -0,0 +1,384 @@
+   * strings generate exceptions.
+   *
+   * @param setter method reference or lambda to set a value of the field being tested on a message
+   *     object
+   * @param getter method reference of lambda to get a value of the field being tested from an
+   *     entity object
+   * @param fieldLabel text identifying the field in {@link
+   *     gov.cms.bfd.pipeline.rda.grpc.source.DataTransformer.TransformationException error
+   *     messages}
+   * @return this object so that calls can be chained
+   */
+  @CanIgnoreReturnValue
+  ClaimTransformerFieldTester<TClaimBuilder, TClaim, TClaimEntity, TTestEntityBuilder, TTestEntity>
+      verifyDateStringFieldTransformedCorrectly(
+          BiConsumer<TTestEntityBuilder, String> setter,
+          Function<TTestEntity, LocalDate> getter,
+          String fieldLabel) {
+    final BiConsumer<TClaimBuilder, String> wrappedSetter =
+        (claimBuilder, value) -> setter.accept(getTestEntityBuilder(claimBuilder), value);
+    final Function<TClaimEntity, LocalDate> wrappedGetter =
+        claim -> getter.apply(getTestEntity(claim));
+    verifyFieldTransformationSucceeds(
+        claimBuilder -> wrappedSetter.accept(claimBuilder, "2021-12-01"),
+        wrappedGetter,
+        LocalDate.of(2021, 12, 1));
+    verifyFieldTransformationFails(
+        claimBuilder -> wrappedSetter.accept(claimBuilder, "not-a-date"),
+        getLabel(fieldLabel),
+        "invalid date");
+    return this;
+  }
+
+  /**
+   * Verifies that an amount field transformation is working properly. This includes verifying that
+   * a value is properly copied from the message object to the entity object and that invalid
+   * decimal strings generate exceptions.
+   *
+   * @param setter method reference or lambda to set a value of the field being tested on a message
+   *     object
+   * @param getter method reference of lambda to get a value of the field being tested from an
+   *     entity object
+   * @param fieldLabel text identifying the field in {@link
+   *     gov.cms.bfd.pipeline.rda.grpc.source.DataTransformer.TransformationException error
+   *     messages}
+   * @return this object so that calls can be chained
+   */
+  @CanIgnoreReturnValue
+  ClaimTransformerFieldTester<TClaimBuilder, TClaim, TClaimEntity, TTestEntityBuilder, TTestEntity>
+      verifyAmountStringFieldTransformedCorrectly(
+          BiConsumer<TTestEntityBuilder, String> setter,
+          Function<TTestEntity, BigDecimal> getter,
+          String fieldLabel) {
+    final BiConsumer<TClaimBuilder, String> wrappedSetter =
+        (claimBuilder, value) -> setter.accept(getTestEntityBuilder(claimBuilder), value);
+    final Function<TClaimEntity, BigDecimal> wrappedGetter =
+        claim -> getter.apply(getTestEntity(claim));
+    verifyFieldTransformationSucceeds(
+        claimBuilder -> wrappedSetter.accept(claimBuilder, "1234.50"),
+        wrappedGetter,
+        new BigDecimal("1234.50"));
```

Comment:
🍰  Could maybe also check that a number with no decimal is properly converted? Something like 56 and/or 56.5. I'd assume the converter should properly be setting the correct scaling to 2 decimals even if not provided.

Reply:
I've allowed the BigDecimal constructor to determine the scale based on the string we receive from the RDA API.  The actual precision when written to the database will be set by the database column definition.

Reply:
Got it. Thanks for explaining!